types.go

```diff
@@ -1045,10 +1012,10 @@ func decodeDateTimeOffset(scale uint8, buf []byte) time.Time {
 		time.FixedZone("", offset*60))
 }
 
-func encodeDateTimeOffset(val time.Time, scale int) (buf []byte) {
+func encodeDateTimeOffset(val time.Time, scale int, loc *time.Location) (buf []byte) {
 	timesize := calcTimeSize(scale)
 	buf = make([]byte, timesize+2+3)
-	days, seconds, ns := dateTime2(val.In(time.UTC))
+	days, seconds, ns := dateTime2(val.In(loc))
```

Comment:
>n(loc)) [](http://example.com/codeflow?start=37&length=7)

I don't think you need to do anything here for DateTimeOffset.
The Go `time.Time` object already has time zone information in it. Also, converting it to a different timezone contradicts the use of `val.Zone()` on line 1023.